types/plugin.d.ts

```diff
@@ -1,12 +1,14 @@
 import { FastifyInstance } from './instance'
 import { RawServerBase, RawRequestDefaultExpression, RawReplyDefaultExpression, RawServerDefault } from './utils'
 
+export type FastifyPluginOptions = Record<string, any>
+
 /**
  * FastifyPluginCallback
  *
  * Fastify allows the user to extend its functionalities with plugins. A plugin can be a set of routes, a server decorator or whatever. To activate plugins, use the `fastify.register()` method.
  */
-export type FastifyPluginCallback<Options extends FastifyPluginOptions = {}, Server extends RawServerBase = RawServerDefault> = (
```

Comment:
Actually here we want to have an empty object as default. I would stick with `{}` and not a generic `Record<string, any>`

Reply:
What about this rule:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md#default-options

> Avoid the Object and {} types, as they mean "any non-nullish value".
This is a point of confusion for many developers, who think it means "any object type".

If you have any input on it

Reply:
Ok seems I've always misused `{}` as default. See this: https://mariusschulz.com/blog/the-object-type-in-typescript#the-empty-type

Reply:
Maybe `Record<never, never>` would do the magic?